app/app.go

```diff
@@ -1063,6 +1066,21 @@ func (app *EthermintApp) GetStoreKey(name string) storetypes.StoreKey {
 	return app.okeys[name]
 }
 
+func (app *EthermintApp) PendingTxStream() <-chan []byte {
+	return app.chPendingTx
+}
+
+func (app *EthermintApp) CheckTx(req *abci.RequestCheckTx) (*abci.ResponseCheckTx, error) {
+	res, err := app.BaseApp.CheckTx(req)
+	if err == nil && res.Code == 0 && req.Type == abci.CheckTxType_New {
+		select {
+		case app.chPendingTx <- req.Tx:
```

Comment:
Better to callback to add to the stream directly, don't need this channel?

Reply:
if we blocking send to the channel, it blocks the mempool when no one consuming.
if we "try send" to the channel like current code, there could lose messages, if it's not consumed in time.

Reply:
I pushed a refactor commit, please review

Reply:
seems still could block in [listener(req.Tx)](https://github.com/crypto-org-chain/ethermint/pull/488/commits/b443d6b7a89fe09d79f5016ab815d5d0cb80de02#diff-0f1d2976054440336a576d47a44a37b80cdf6701dd9113012bce0e3c425819b7R1079)

Reply:
it don't block